app/models/user.rb

```diff
@@ -63,9 +63,8 @@ def previous_upload_for(category)
   end
 
   def pbs
-    runs.where(
-      id: categories.map { |category| pb_for(category).id10 } | runs.where(category: nil).pluck(:id)
-    )
+    Run.where(user: self).select('DISTINCT ON (category_id) *').order('category_id, realtime_duration_ms ASC')
```

Comment:
Any reason for using `Run.where(user: self)` here over `runs`?

Reply:
To be honest, when I started testing stuff I was working with `joins` which needs to be done on `User`, and then got stuck in that mindset.  I'll take a look and see if this and the one below are equivalent.

Reply:
Doing it like you said, i was able to add an extra where clause to make it so there will not be any duplicates, and then use `union_all` which doesn't have a de-duplication step which has a bit better performance.